### PR TITLE
Release tool: new skip_edit_download_script flag and better approach to add new line before eof

### PR DESCRIPTION
**Pull request by chxchx**
**Release note**:

```release-note
none
```

### CODE REVIEW

File: toolbox/githubctl/main.go [hunk 1]
```diff
@@ -24,14 +24,15 @@ import (
...
-	nextRelease = flag.String("next_release", "", "Tag of the next release")
-	githubClnt  *u.GithubClient
+	owner                  = flag.String("owner", "istio", "Github owner or org")
+	tokenFile              = flag.String("token_file", "", "File containing Github API Access Token")
+	op                     = flag.String("op", "", "Operation to be performed")
+	repo                   = flag.String("repo", "", "Repository to which op is applied")
+	baseBranch             = flag.String("base_branch", "", "Branch to which op is applied")
+	refSHA                 = flag.String("ref_sha", "", "Reference commit SHA used to update base branch")
+	nextRelease            = flag.String("next_release", "", "Tag of the next release")
+	skipEditDownloadScript = flag.Bool("skip_edit_download_script", false, "Have download script point to old version")
```
Comment (ldemailly) [inline toolbox/githubctl/main.go]: I'd make the default to not edit that script.

The idea there is you make the release first - then spend some days making sure it's good/better than the previous one and ready for prime time - then update the download script

so I see that automatic updates as very rare - also that file is special as in there is only 1 so while once we cut the 0.2 branch, we will change istio.RELEASE in the branch, the download change is on master,...

File: toolbox/util/commonUtils.go [hunk 2]
```diff
@@ -36,6 +36,9 @@ func ReadFile(filePath string) (string, error) {
 
 // WriteFile overwrites the file on the given path with content
 func WriteFile(filePath, content string) error {
```
Comment (ldemailly) [inline toolbox/util/commonUtils.go]: nit: might need to be changed to WriteTextFile if you assume you can add a \n
or are you sure it's never used to write binary data? (I guess it takes a string and not bytes so maybe that's implicit)

File: toolbox/util/commonUtils.go [hunk 3]
```diff
@@ -36,6 +36,9 @@ func ReadFile(filePath string) (string, error) {
...
+	if len(content) == 0 || content[len(content)-1] != '\n' {
```
Comment (ldemailly) [inline toolbox/util/commonUtils.go]: are you sure you want to change 0 bytes file (which is fine, sometimes marker files are like that) into 1 byte file for the new line ?

File: toolbox/util/commonUtils.go [hunk 4]
```diff
@@ -36,6 +36,9 @@ func ReadFile(filePath string) (string, error) {
...
+		content += "\n"
```
Comment (ldemailly) [inline toolbox/util/commonUtils.go]: yeah ! fix all the files ! (not perf critical)

File: toolbox/util/commonUtils.go (hunk 3 above)
Comment (chxchx) [reply toolbox/util/commonUtils.go]: This is a good point.

File: toolbox/util/commonUtils.go (hunk 2 above)
Comment (chxchx) [reply toolbox/util/commonUtils.go]: Renaming to `WriteTextFile` totally makes sense. It is just difficult to know all the places this function has been used and also refactor every such occurrence. Maybe we could leave this text input as implicit defined as you have pointed out, and create a `WriteBinaryFile` when we need it.

File: toolbox/githubctl/main.go (hunk 1 above)
Comment (chxchx) [reply toolbox/githubctl/main.go]: Got it. Thanks for the detailed explanation!

File: toolbox/util/commonUtils.go (hunk 2 above)
Comment (ldemailly) [reply toolbox/util/commonUtils.go]: looks like it's used exactly once so far:
https://github.com/istio/test-infra/search?l=Go&q=WriteFile&type=&utf8=%E2%9C%93
so I would rename it now :-)

File: toolbox/githubctl/main.go [hunk 5]
```diff
@@ -24,14 +24,15 @@ import (
...
-	nextRelease = flag.String("next_release", "", "Tag of the next release")
-	githubClnt  *u.GithubClient
+	owner                = flag.String("owner", "istio", "Github owner or org")
+	tokenFile            = flag.String("token_file", "", "File containing Github API Access Token")
+	op                   = flag.String("op", "", "Operation to be performed")
+	repo                 = flag.String("repo", "", "Repository to which op is applied")
+	baseBranch           = flag.String("base_branch", "", "Branch to which op is applied")
+	refSHA               = flag.String("ref_sha", "", "Reference commit SHA used to update base branch")
+	nextRelease          = flag.String("next_release", "", "Tag of the next release")
+	udpateDownloadScript = flag.Bool("update_download_script", false, "Have download script point to latest version")
```
Comment (ldemailly) [inline toolbox/githubctl/main.go]: thanks for flipping the flag around!

typo: udpate -> update
probably should turn on the spell checker linter on this repo

File: toolbox/githubctl/main.go [hunk 6]
```diff
@@ -204,16 +206,18 @@ func CreateIstioReleaseUploadArtifacts() error {
...
 		}
-		if err := u.WriteFile(releaseTagFile, *nextRelease+"\n"); err != nil {
+		if err := u.WriteFile(releaseTagFile, *nextRelease); err != nil {
 			return err
 		}
-		return u.UpdateKeyValueInFile(
-			downloadScript, "ISTIO_VERSION", releaseTag)
+		if *udpateDownloadScript {
+			log.Printf("Updating download script with latest release")
+			return u.UpdateKeyValueInFile(
```
Comment (ldemailly) [inline toolbox/githubctl/main.go]: why return ?
also probably want to prBody += " and updating download script"

Comment (sebastienvas) [reply toolbox/githubctl/main.go]: he is doing a func inside a func.

Comment (ldemailly) [reply toolbox/githubctl/main.go]: oic, missed that, thx

File: toolbox/util/commonUtils.go (hunk 2 above)
Comment (chxchx) [reply toolbox/util/commonUtils.go]: Done

File: toolbox/githubctl/main.go (hunk 5 above)
Comment (chxchx) [reply toolbox/githubctl/main.go]: Done

**Comment by chxchx** [comment]
PTAL

**Review by ldemailly — APPROVED** [review]
/lgtm
thank you

**Review by sebastienvas — APPROVED** [review]

**Pull request merged**